src/Countries/Ecuador.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Ecuador extends Country
+{
+    public function countryCode(): string
+    {
+        return 'ec';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Año Nuevo' => '01-01',
+        ], $this->variableHolidays($year));
+    }
+
+    public function nearestDay(int $year, int $month, int $day)
+    {
+        $date = CarbonImmutable::createFromDate($year, $month, $day);
+
+        if($date->is('Tuesday') || $date->is('Saturday')) {
+            return $date->subDay();
+        }
+
+        if($date->is('Sunday')) {
+            return $date->addDay();
+        }
+
+        if($date->is('Wednesday') || $date->is('Thursday')) {
+            return $date->next(CarbonImmutable::FRIDAY);
+        }
+
+        return $date;
+    }
+
+    public function getChristmasHoliday(int $year)
+    {
+        $date = CarbonImmutable::createFromDate($year, 12, 25);
+
+        if($year === 2022) {
+            return $date->addDay();
+        }
+
+        return $date;
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = $this->easter($year);
```

Comment:
Instead of mixing two languages, I suggest to use English here and implement `HasTranslations` & `Translatable`.

Reply:
ok, I just made the changes to use translations